packages/ui/src/views/repo/pull-request/components/commit-selector/commit-selector-dropdown.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useMemo, useState } from 'react'
+
+import { DropdownMenuContent, DropdownMenuItem, Icon, SearchBox } from '@/components'
+import { cn } from '@utils/cn'
+
+import { CommitSelectorListItem } from '../../compare/components/types'
+import { CommitSelectorDropdownProps } from '../../types'
+
+const filterItems = (items: CommitSelectorListItem[], query: string) => {
+  if (!query.trim()) return items
+
+  return items.filter(item => item.title.toLowerCase().includes(query.toLowerCase().trim()))
+}
+
+export const CommitSelectorDropdown: FC<CommitSelectorDropdownProps> = ({
+  selectedCommit,
+  onSelectCommit,
+  commitList,
+
+  useTranslationStore
+}) => {
+  const [searchQuery, setSearchQuery] = useState('')
+  const { t } = useTranslationStore()
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value)
+  }
+  const filteredItems = useMemo(() => {
+    return filterItems(commitList, searchQuery)
+  }, [commitList, searchQuery])
+
+  return (
+    <DropdownMenuContent className="w-[298px] p-0" align="start">
+      <div className="px-3 pt-2">
+        <SearchBox.Root
+          className="mt-[18px] w-full"
```

Comment:
why do we need custom `[18px]` spacing? searchbox should be standardised

Reply:
fixed